content/docs/Guides-CustomConnector.md

```diff
@@ -0,0 +1,180 @@
+---
+title: CustomConnector
+date: "2018-07-09"
+---
+
+To create a custom connector, you must implement following classes: 
+
+- `CustomConnector` 
+- `CustomContext` 
+- `CustomEvent` 
+- `CustomClient` (optional) 
+
+## CustomEvent
+
+Event is a parser for the platform raw event. For example, if text messages have following format on the custom platform:
+
+```js
+{
+  user: {
+    id: '123456',
+  },
+  message: {
+    text: 'Hello, world',
+  },
+}
+```
+
+You can write a `CustomEvent` class to hide the underlying complex raw event detail:
+
+```js
+class CustomEvent {
+  constructor(rawEvent) {
+    this._rawEvent = rawEvent;
+  }
+
+  get rawEvent() {
+    return this._rawEvent;
+  }
+
+  get isText() {
+    return !!(this._rawEvent.message && this._rawEvent.message.text)
+  }
+
+  get text() {
+    if (this.isText) {
+      return this._rawEvent.message.text;
+    }
+    return null;
+  }
+}
+```
+
+Following getters are commonly cross-platform implemented:
+
+- `rawEvent`
+- `isMessage`
+- `message`
+- `isText`
+- `text`
+- `isPayload`
+- `payload`
+
+Free free to implement platform only getter on it.
+
+```js
+class CustomEvent {
+  // ...
+  get isSuperCoolSomething () {}
+}
+```
+
+## CustomContext
+
+Context is an abstraction of current conversation context. Every context instance includes event, session, state, and all available context methods:
+
+```js
+{
+  event,
+  session,
+  state,
+  ...contextMethods,
+}
+```
+
+You can write those context side effect methods directly or wrap another `CustomClient` class to do the job.
+
+```js
+class CustomContext extends Context {
+  constructor({
+    event,
+    session,
+    initialState,
+  }) {
+    super({ event, session, initialState });
+  }
+
+  get platform() {
+    return 'custom';
+  }
+
+  async sendText(text) {
+    // ...
+  }
+
+  async typing(milliseconds) {
+    // ...
+  }
+}
+```
+
+Following methods are commonly cross-platform implemented:
+
+- `sendText`
+- `typing`
+
+## CustomConnector
+
+Connector is a bridge between bot and platform. You need to implement following methods to tell bot how to find session key from request and create conversation context.
+
+- getUniqueSessionKey
+
+Find unique session identifier, for example, user id, group id, channel id...and so on.
```

Comment:
I think it is good to use h3 or h4 heading on `getUniqueSessionKey` method name.
Or maybe we can add a indent list on the description.

It will enhance the readability.